CMakeLists.txt

```diff
@@ -542,66 +533,48 @@ else ()
 endif ()
 
 if (CLR_CMAKE_TARGET_ARCH_AMD64)
-  if (CLR_CMAKE_PLATFORM_UNIX)
-    add_definitions(-DDBG_TARGET_AMD64_UNIX)
-  endif()
-  add_definitions(-D_TARGET_64BIT_=1)
-  add_definitions(-D_TARGET_AMD64_=1)
-  add_definitions(-DDBG_TARGET_64BIT=1)
-  add_definitions(-DDBG_TARGET_AMD64=1)
-  add_definitions(-DDBG_TARGET_WIN64=1)
+  add_definitions(-DTARGET_AMD64=1)
+  add_definitions(-DTARGET_64BIT=1)
 elseif (CLR_CMAKE_TARGET_ARCH_ARM64)
-  if (CLR_CMAKE_PLATFORM_UNIX)
-    add_definitions(-DDBG_TARGET_ARM64_UNIX)
-  endif()
-  add_definitions(-D_TARGET_ARM64_=1)
-  add_definitions(-D_TARGET_64BIT_=1)
-  add_definitions(-DDBG_TARGET_64BIT=1)
-  add_definitions(-DDBG_TARGET_ARM64=1)
-  add_definitions(-DDBG_TARGET_WIN64=1)
+  add_definitions(-DTARGET_ARM64=1)
+  add_definitions(-DTARGET_64BIT=1)
   add_definitions(-DFEATURE_MULTIREG_RETURN)
 elseif (CLR_CMAKE_TARGET_ARCH_ARM)
-  if (CLR_CMAKE_PLATFORM_UNIX)
-    add_definitions(-DDBG_TARGET_ARM_UNIX)
-  elseif (WIN32 AND NOT DEFINED CLR_CROSS_COMPONENTS_BUILD)
+  if (CLR_CMAKE_HOST_WIN32 AND NOT DEFINED CLR_CROSS_COMPONENTS_BUILD)
     # Set this to ensure we can use Arm SDK for Desktop binary linkage when doing native (Arm32) build
     add_definitions(-D_ARM_WINAPI_PARTITION_DESKTOP_SDK_AVAILABLE=1)
     add_definitions(-D_ARM_WORKAROUND_)
-  endif (CLR_CMAKE_PLATFORM_UNIX)
-  add_definitions(-D_TARGET_ARM_=1)
-  add_definitions(-DDBG_TARGET_32BIT=1)
-  add_definitions(-DDBG_TARGET_ARM=1)
+  endif (CLR_CMAKE_HOST_WIN32 AND NOT DEFINED CLR_CROSS_COMPONENTS_BUILD)
+  add_definitions(-DTARGET_ARM=1)
 elseif (CLR_CMAKE_TARGET_ARCH_I386)
-  add_definitions(-D_TARGET_X86_=1)
-  add_definitions(-DDBG_TARGET_32BIT=1)
-  add_definitions(-DDBG_TARGET_X86=1)
+  add_definitions(-DTARGET_X86=1)
 else ()
   clr_unknown_arch()
 endif (CLR_CMAKE_TARGET_ARCH_AMD64)
 
-if(WIN32)
+if(CLR_CMAKE_HOST_WIN32)
+  add_definitions(-DHOST_WINDOWS)
   add_definitions(-DWIN32)
   add_definitions(-D_WIN32)
   add_definitions(-DWINVER=0x0602)
   add_definitions(-D_WIN32_WINNT=0x0602)
   add_definitions(-DWIN32_LEAN_AND_MEAN=1)
   add_definitions(-D_CRT_SECURE_NO_WARNINGS)
-endif(WIN32)
+endif(CLR_CMAKE_HOST_WIN32)
 
 #--------------------------------------
 # FEATURE Defines
 #--------------------------------------
 
 add_definitions(-DFEATURE_CORESYSTEM)
 
-if(CLR_CMAKE_PLATFORM_UNIX)
-    add_definitions(-DFEATURE_PAL)
+if(CLR_CMAKE_HOST_UNIX)
```

Comment:
In the `runtime`, the feature defines are now set by target OS and properties....  I don't anticipate needing that here...  So I didn't do that yet...  But if we need to copy code freely between `runtime` and `diagnostics` repo maybe I should...

One difference is that in the `runtime`, I mapped `FEATURE_PAL` to `TARGET_UNIX` by default....